src/python/WMCore/WorkQueue/DataStructs/WorkQueueElement.py

```diff
@@ -71,8 +71,12 @@ def __init__(self, **kwargs):
         self.setdefault('OpenForNewData', False)
         # When was the last time we found new data (not the same as when new data was split), e.g. An open block was found
         self.setdefault('TimestampFoundNewData', 0)
-        # Should we check the location of the inputs, or trust the initial values?
+        # Trust initial input and pileup location or not
         self.setdefault('NoLocationUpdate', False)
```

Comment:
Hi Alan, we have to remove completely 'NoLocationUpdate', or keep this line.Otherwise it fails in line 277.
I am fine with completely removing it. But I think you left it other places for backward compatibility. So I think we need to keep 'NoLocationUpdate' in wmagent branch.